recipes/hdf5/all/conanfile.py

```diff
@@ -72,6 +73,11 @@ def validate(self):
         if hasattr(self, "settings_build") and tools.cross_building(self, skip_x64_x86=True):
             # While building it runs some executables like H5detect
             raise ConanInvalidConfiguration("Current recipe doesn't support cross-building (yet)")
+        if self.options.parallel:
+            if self.options.enable_cxx:
+                raise ConanInvalidConfiguration("Parallel and C++ options are mutually exclusive")
```

Comment:
this is unfortunate (?)

Reply:
Yes the H5CPP library and HDF5 Parallel are an "untested configuration", so HDF5 makes them mutually exclusive by default - though it is possible to use them together in an "at your own risk" configuration.

It is possible however to use HDF5 as a low-level implementation to support higher level C++ implementations (ie, https://github.com/conan-io/conan-center-index/pull/6239). Hence my interest in this PR :)